### ISSUE TITLE
Add support to Parcelable

### ISSUE THREAD

**Issue by StErMi** [kaaes/spotify-web-api-android#40]
Current models as far as I can see are not implementing Parcelable.
This is problem if I would like to pass those models between activities or Service.

Do you plan to add it?

**Comment by kaaes** [comment]
I think it would be a good idea to add it.
I can look into this when I have a bit of time but please feel free to open a PR if you don't want to wait.

**Comment by JohnnyCrazy** [comment]
if I'm not mistaken, http://developer.android.com/reference/android/os/Parcelable.html would require a lot of adjustments in each model and would furthermore change the simplicity of Retrofit models.

But I'm not sure, maybe there is some nice way of implementing this, without adding a lot of stuff to each model

(As a little hacky workaround, you could always parse the model to a JSON string and pass it to a new activity and then revert it to the model)

**Comment by StErMi** [comment]
You will double the effort becuase, as far as I know, Retrofit are already
parsing that JSON the first time with gson lib (as default) to map the
API's respons (in json) to the model.


I don't think that this will add a complexity to Retrofit, it just need a
lot of boiler code.


Implementing Parcelable will for sure change the project structure, but
this is wrapper for android and not java ;)

**Comment by StErMi** [comment]
@kaaes I can implement it on my own ( I need it for my current project ) so it should not be a real pain for me. 
But I think that with my modification (it will be an android library and not a simple java plugin) I will break your current gradle build

**Comment by kaaes** [comment]
Maybe there's a tool that would help cut the amount of boilerplate code.
Super quick google search yielded this: https://github.com/johncarl81/parceler. I don't know how good it is but perhaps it's a direction worth exploring.

@StErMi it would be interesting to see how this project would need to change if this functionality was implemented. I am not against changing it to be an android library if there's a good reason for that. Let's keep this conversation going?

**Comment by StErMi** [comment]
The first good reason to be an android library is that I can implements Parcelable interface on those models. This will allow us to transfer those models between activies and services inside bundle.

**Comment by ryandt** [comment]
I took a break from my Spotify project for a couple months due to some other commitments, and the first thing I started back up on last night was dealing with this exact issue. I'm happy to see some others expressing the same need.

Parcelable models should be fundamentally implied through the name of the library. If 'spotify-web-api-android' is marketed as an Android library, developers looking for optimal performance are going to expect Parcelable model objects.

I ran into the Parceler library last night that @kaaes mentioned above. It looks like a code generator fueled on annotations. Has anyone taken Parceler for a test run or started any work on this issue yet? If not, I can maybe dig into it sometime this weekend or next next week if I get the time.
